### DISCUSSION TITLE
Support for Tailwind v3.2

### DISCUSSION THREAD

**Discussion by brandonmcconnell** [dcastil/tailwind-merge#157] [Help] [answered]
The Tailwind v3.2 beta is out, and the syntax is [fairly finalized](https://github.com/tailwindlabs/tailwindcss/discussions/9574). Just wanted to start a discussion to start any needed discussion surrounding supporting this new version.

The details of the new changes with Tailwind v3.2 can be found here: https://github.com/tailwindlabs/tailwindcss/discussions/9574

Notable highlights:

* Breakpoint ranges by stacking new `max` breakpoints on existing one, e.g.:
  * `p-8 lg:p-0 → max-lg:p-8`
  * `md:sr-only xl:not-sr-only → md:max-xl:sr-only`
* Arbitrary breakpoints, also works with stacked `max` breakpoints ☝🏼, e.g.:
  * `min-[712px]:max-[877px]:right-16`
* Dynamic `group-[...]` and `peer-[...]` variants, e.g.:
  * `group-[:nth-of-type(3)_&]:border-black`
* Parameterized variants with `matchVariant`, e.g.:
  * `v => &[placement=${v}], values: { t: "top", r: "right", b: "bottom", l: "left", }`
* Nested `group` and multiple `peer` support using variant modifiers, e.g.:
  * `group/sidebar → group-hover/sidebar:opacity-75`
* New variant for `@supports` rules, e.g.:
  * `supports-[display:grid]:grid`
  New variants for `aria-*` attributes , e.g.:
  * `<div aria-checked="true" class="aria-checked:bg-blue-600">`
  New variants for `data-*` attributes, e.g.:
  * `<div data-size="large" class="data-[size=large]:p-8">`
* Dynamic `group-[...]` and `peer-[...]` variants, e.g.:
  * `peer + peer-[:local-link]:ml-3`
  * `group > group-[:nth-of-type(3)_&]:border-black`
* Multiple config files in one project using `@config`, e.g.:
  * `@config "./tailwind.admin.config.js"`

**Comment by dcastil** [comment, marked as answer]
What a timing, Tailwind released [v3.2.0](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.2.0) just 2 hours after you posted. 😄

I'll start with this as soon as possible, hopefully support will be there in a few days. When I have a PR ready, I'll post it here.

**Reply by brandonmcconnell** [reply]
Awesome, thanks. 👏🏼

I'll wait on your update before updating our local codebase. I'd be happy to share any findings I have once I update.

**Reply by dcastil** [reply]
I have a PR ready → https://github.com/dcastil/tailwind-merge/pull/159

@brandonmcconnell feel free to check it out and review if you want. I'll go to sleep now and will merge it tomorrow.

**Reply by brandonmcconnell** [reply]
Looks great! Thanks for the merge 🙌🏼
